api/proxy/servers/arbitrum_altda/handlers.go

```diff
@@ -105,31 +109,75 @@ type Handlers struct {
 	//       We should dig into this underlying logging and see if there's a way to intuitively override, disable,
 	//       or enforce consistency between log outputs.
 
+	log            logging.Logger
 	eigenDAManager *store.EigenDAManager
 }
 
-func NewHandlers(m *store.EigenDAManager) IHandlers {
+// NewHandlers is a constructor
+func NewHandlers(m *store.EigenDAManager, l logging.Logger) IHandlers {
 	return &Handlers{
+		log:            l,
 		eigenDAManager: m,
 	}
 }
 
-// GetSupportedHeaderBytes determines whether or not the sequencer message header byte is an EigenDAV2 cert type.
-// Arbitrum Nitro does this check via a bitwise AND which can cause overlapping and requires careful future
-// management. while we could determine a byte value with bits that don't overlap - it's more maintainable
-// to do a literal comparison and assume OCL NOR our competitors would never introduce a conflicting byte value
+// GetSupportedHeaderBytes returns the supported DA Header bytes by the CustomDA server
+// this method is designed to return a span of bytes for compatibility with
+// Arbitrum AnyTrust where multiple message types are supported.
+// For CustomDA the provider only returns the Arbitrum CustomDA header byte.
 func (h *Handlers) GetSupportedHeaderBytes(ctx context.Context) (*SupportedHeaderBytesResult, error) {
+	h.logMethodCall(MethodGetSupportedHeaderBytes)
+
 	return &SupportedHeaderBytesResult{
-		HeaderBytes: []byte{EigenDAV2MessageHeaderByte},
+		HeaderBytes: []byte{commitments.ArbCustomDAHeaderByte},
 	}, nil
 }
 
+// deserializeCertFromSequencerMsg reads the VersionedCert from the raw sequencer message provided
+// by the DA Client
+func (h *Handlers) deserializeCertFromSequencerMsg(sequencerMsg hexutil.Bytes) (certs.VersionedCert, error) {
+	if len(sequencerMsg) <= 42 {
+		return certs.VersionedCert{},
+			fmt.Errorf("sequencer message expected to be >=42 bytes, got: %d", len(sequencerMsg))
+	}
+
+	seqMessageWithoutHeader := sequencerMsg[40:]
+
+	daHeaderByte := seqMessageWithoutHeader[0]
+	if daHeaderByte != commitments.ArbCustomDAHeaderByte {
+		return certs.VersionedCert{},
+			fmt.Errorf("expected CutomDAHeader byte (%x) for 0th index byte of message, instead got: %x ",
+				commitments.ArbCustomDAHeaderByte, daHeaderByte)
+	}
+
+	daLayerByte := seqMessageWithoutHeader[1]
+	if daLayerByte != commitments.EigenDALayerByte {
+		return certs.VersionedCert{},
+			fmt.Errorf("expected EigenDALayer byte (%x) for 1st index byte of message, instead got: %x ",
+				commitments.EigenDALayerByte, daHeaderByte)
```

Comment:
iiuc this error message references `daHeaderByte` instead of `daLayerByte` in the format string

Reply:
[379a7d4](https://github.com/Layr-Labs/eigenda/pull/2218/commits/379a7d4c194d471b9ffc52b88a3905d47a2659ea)